javalib/src/main/scala/java/net/Socket.scala

```diff
@@ -0,0 +1,304 @@
+package java.net
+
+import java.io.{InputStream, OutputStream, IOException, Closeable}
+
+class Socket protected (private[net] val impl: SocketImpl,
+                        private[net] var addr: InetAddress,
+                        private[net] var port: Int,
+                        private[net] var localAddr: InetAddress,
+                        private[net] var localPort: Int,
+                        streaming: Boolean,
+                        shouldStartup: Boolean)
+    extends Closeable {
+
+  private[net] var created        = false
+  private[net] var bound          = false
+  private[net] var connected      = false
+  private[net] var closed         = false
+  private[net] var inputShutdown  = false
+  private[net] var outputShutdown = false
+
+  if (!streaming) {
+    throw new UnsupportedOperationException(
+      "Only streaming sockets are supported " +
+        "in this class for the moment.")
+  }
+
+  if (shouldStartup) {
+    startup(addr, port)
+  }
+
+  private def startup(dstAddr: InetAddress, dstPort: Int, timeout: Int = 0) = {
+    if (dstPort < 0 || dstPort > 65535)
+      throw new IllegalArgumentException(
+        "Socket port must be between 0 and 65535")
+
+    impl.create(streaming)
+    val created = true
```

Comment:
I guess the socket descriptor was just created twice if the function `checkClosedAndCreate` was called :D Fixed